packages/core/components/NestedDropdown/NestedDropdown.tsx

```diff
@@ -0,0 +1,251 @@
+import { useState, useEffect, useRef, useMemo } from 'react'
+import './nesteddropdown.styles.css'
+
+const Options: React.FC<{
+  currentOptions: (string | number)[]
+  label: string
+  handleSecondTierSelect: Function
+  userSelectedTierTwoLabel: string
+  userSearchTerm: string
+}> = ({ currentOptions, label, handleSecondTierSelect, userSelectedTierTwoLabel, userSearchTerm }) => {
+  const [isTierOneExpanded, setIsTierOneExpanded] = useState(true)
+
+  useEffect(() => {
+    setIsTierOneExpanded(userSearchTerm.length > 0 ? true : isTierOneExpanded)
+  }, [userSearchTerm])
+
+  const handleGroupClick = e => {
+    const leaveExpanded = e.target.className === 'selectableItem' ? true : !isTierOneExpanded
+    setIsTierOneExpanded(leaveExpanded)
+  }
+
+  const handleSecondTierClick = (tierTwo: string | number) => {
+    handleSecondTierSelect(tierTwo)
+  }
+
+  const handleKeyUp = (e: { target: any; key: string }) => {
+    const currentItem = e.target
+    if (e.key === 'ArrowRight') setIsTierOneExpanded(true)
+    else if (e.key === 'ArrowLeft') {
+      if (currentItem.className === 'selectableItem') currentItem.parentNode.parentNode.focus()
+      setIsTierOneExpanded(false)
+    } else if (e.key === 'Enter') {
+      currentItem.className === 'selectableItem' ? handleSecondTierClick(currentItem.dataset.value) : setIsTierOneExpanded(!isTierOneExpanded)
+    }
+  }
+
+  return (
+    <>
+      <li role='treeitem' key={label} tabIndex={0} aria-label={label} onClick={handleGroupClick} onKeyUp={handleKeyUp} className={isTierOneExpanded ? 'nested-dropdown__group' : 'nested-dropdown__group--hidden'}>
+        <span id={label}>{label}</span> {<span aria-hidden='true'> {isTierOneExpanded ? <>&#9662;</> : <>&#9652;</>}</span>}
+        <ul className='' aria-expanded={isTierOneExpanded} role='group' tabIndex={-1} aria-labelledby={label}>
```

Comment:
After doing some research and trial and error, I think the UL>LI setup for the NestedDropdown is the best options. The optgroup has preset attributes that I would need to code around to make it work. I tried a bit and wasn't successful. I may be able to do it but I didn't want to spend too much time on it. This may be something we can look into in the future. But I think the time investment, at least on my part, would be too much